fabric-screen-api-v1/src/main/java/net/fabricmc/fabric/api/client/screen/v1/ScreenEvents.java

```diff
@@ -0,0 +1,150 @@
+
+@Environment(EnvType.CLIENT)
+public final class ScreenEvents {
+	/**
+	 * An event that is called when a {@link Screen#init(MinecraftClient, int, int) screen is initialized}.
+	 */
+	public static final Event<Init> INIT = EventFactory.createArrayBacked(Init.class, callbacks -> (client, screen, context, scaledWidth, scaledHeight) -> {
+		for (Init callback : callbacks) {
+			callback.onInit(client, screen, context, scaledWidth, scaledHeight);
+		}
+	});
+
+	/**
+	 * An event that is called before a screen is rendered.
+	 */
+	public static final Event<BeforeRender> BEFORE_RENDER = EventFactory.createArrayBacked(BeforeRender.class, callbacks -> (client, matrices, screen, context, mouseX, mouseY, tickDelta) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			final Profiler profiler = client.getProfiler();
+			profiler.push("fabricRenderScreen");
+
+			for (BeforeRender callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.beforeRender(client, matrices, screen, context, mouseX, mouseY, tickDelta);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (BeforeRender callback : callbacks) {
+				callback.beforeRender(client, matrices, screen, context, mouseX, mouseY, tickDelta);
+			}
+		}
+	});
+
+	/**
+	 * An event that is called after a screen is rendered.
+	 */
+	public static final Event<AfterRender> AFTER_RENDER = EventFactory.createArrayBacked(AfterRender.class, callbacks -> (client, matrices, screen, context, mouseX, mouseY, tickDelta) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			final Profiler profiler = client.getProfiler();
+			profiler.push("fabricRenderScreen");
+
+			for (AfterRender callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.afterRender(client, matrices, screen, context, mouseX, mouseY, tickDelta);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (AfterRender callback : callbacks) {
+				callback.afterRender(client, matrices, screen, context, mouseX, mouseY, tickDelta);
+			}
+		}
+	});
+
+	/**
+	 * An event that is called before a screen is ticked.
+	 */
+	public static final Event<BeforeTick> BEFORE_TICK = EventFactory.createArrayBacked(BeforeTick.class, callbacks -> (client, screen, context) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			final Profiler profiler = client.getProfiler();
+			profiler.push("fabricScreenTick");
+
+			for (BeforeTick callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.beforeTick(client, screen, context);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (BeforeTick callback : callbacks) {
+				callback.beforeTick(client, screen, context);
+			}
+		}
+	});
+
+	/**
+	 * An event that is called after a screen is ticked.
+	 */
+	public static final Event<AfterTick> AFTER_TICK = EventFactory.createArrayBacked(AfterTick.class, callbacks -> (client, screen, context) -> {
```

Comment:
Whats the use case for the tick events?

Reply:
I'd argue it's nessecary if someone adds a widget to a screen that needs to be ticked.

For example if someone added a sort of chyron to the title screen showing news about the latest snapshots, they would need to tick that element of the screen to advance the position of text.

Reply:
So in case a tick-needed widget added to a non-ticking screen?

Reply:
All screens are ticking, but child elements of a screen are not automatically ticking (some screen impls do do that).

Often the tick method of each element is called in the screen's tick method